mpf/platforms/virtual.py

```diff
@@ -311,23 +320,41 @@ class VirtualSegmentDisplay(SegmentDisplayPlatformInterface):
 
     """Virtual segment display."""
 
-    __slots__ = ["text", "flashing", "colors"]
+    __slots__ = ["text", "flashing", "platform_options", "colors", "machine", "post_update_events"]
 
-    def __init__(self, number) -> None:
+    def __init__(self, number, platform_options, machine) -> None:
         """Initialise virtual segment display."""
         super().__init__(number)
+        self.machine = machine
         self.text = ''
         self.flashing = FlashingType.NO_FLASH
-        self.colors = 'FFFFFF'
+        self.colors = [RGBColor('FFFFFF')]
+        self.platform_options = platform_options
+        if self.platform_options and 'post_update_events' in self.platform_options:
```

Comment:
Why do we want to post events when updating a segment_display? That sounds like a job for event_player to me.

Reply:
This was my not so elegant solution to communicating with the MC. My stumbling point was that BCP was not setup and connected when the hardware was being setup so this was an easy way to solve that timing issue. I'm happy to get rid of this and send the trigger using bcp directly bypassing the need for any event generation.